bzt/engine.py

```diff
@@ -1023,27 +1023,37 @@ def get_requests(self):
         """
         Generator object to read requests
         """
-        scenario = self
-        requests = scenario.get("requests", [])
+        requests = self.get("requests", [])
         for key in range(len(requests)):
             req = ensure_is_dict(requests, key, "url")
-            res = namedtuple("HTTPReq",
-                             ('url', 'label', 'method', 'headers', 'timeout', 'think_time', 'config', "body"))
-            url = req.get("url", ValueError("Option 'url' is mandatory for request"))
-            label = req.get("label", url)
-            method = req.get("method", "GET")
-            headers = req.get("headers", {})
-            timeout = req.get("timeout", None)
-            think_time = req.get("think-time", None)
-
-            body = None
-            bodyfile = req.get("body-file", None)
-            bodyfile_path = self.engine.find_file(bodyfile)
-            if bodyfile_path:
+            yield HTTPRequest(config=req, engine=self.engine)
+
+
+class Request(object):
+    def __init__(self, config):
+        self.config = config
+
+
+class HTTPRequest(Request):
+    def __init__(self, config, engine):
+        super(HTTPRequest, self).__init__(config)
+        self.engine = engine
```

Comment:
why pass engine and not logger? I'd minimize relationship to engine from class if it just needs logger